cmd/thanos/sidecar.go

```diff
@@ -212,10 +231,13 @@ func runSidecar(
 	}
 
 	{
+		// todo(someshkoli): Not sure how to properly implement RoundTripper for these properties or
```

Comment:
@Namanl2001 please have a look at this :thinking: I guess there's no clean way to set these properties. (might need to rewrite auth for promclient or make some changes thanoshttp for custom transporter)

Reply:
I think we need some 'generic' transporter that we can re-use everywhere. I had the same 'issue': https://github.com/thanos-io/thanos/pull/3970/files/e34e60e2848f2233809d67d5398151b670341ea1#diff-95c18d923c79bf32e4f8b3370d9a925aa910643559d351fab6bc307691c8f672R119 

The thanoshttp has one func which is:
```
// NewTransport creates a new http.Transport with default settings.
```
I guess we could add an other function that has httpConfig as argument for re-usability in the entire project.

Reply:
> I think we need some 'generic' transporter that we can re-use everywhere. I had the same 'issue': https://github.com/thanos-io/thanos/pull/3970/files/e34e60e2848f2233809d67d5398151b670341ea1#diff-95c18d923c79bf32e4f8b3370d9a925aa910643559d351fab6bc307691c8f672R119
> 
> The thanoshttp has one func which is:
> 
> ```
> // NewTransport creates a new http.Transport with default settings.
> ```
> 
> I guess we could add an other function that has httpConfig as argument for re-usability in the entire project.

Agreed, this is being used at 2-3 i guess and making a generic one can improve reusablity. @wiardvanrij I hope the overall idea in implementation is correct ? I can move forward with fixing this transporter issue ?

Reply:
I'm not the right person to make a definitive review. I miss the exact technical knowledge on it. However I was at the meeting and I think you made what was discussed. Also it looks logical and makes sense. 

You could always wait for a review if you think you might invest a lot of time into a feature you are unsure about. However if you have a good feeling about it, go make awesome stuff and continue 👍

Reply:
Alright thanks

Reply:
I think this transport and connection flags 

```
conf.connection.maxIdleConnsPerHost
conf.connection.maxIdleConns
```

Are colliding with out `promethues.http-config` flag. The idea would be to unify those? E.g Deprecating connection flags if that makes sense? Or reusing them? 

The goal is to have one way of configuring HTTP configuration.

Reply:
Let's remove

>   --receive.connection-pool-size=RECEIVE.CONNECTION-POOL-SIZE
                                 Controls the http MaxIdleConns. Default is 0,
                                 which is unlimited
      --receive.connection-pool-size-per-host=100

And use our prometheus flags.